stablehlo/dialect/TypeInference.cpp

```diff
@@ -1344,47 +1344,60 @@ LogicalResult inferAllToAllOp(
   return success();
 }
 
-LogicalResult inferBatchNormGradOp(
```

Comment:
Agreed with Smit about deduplicating into a single helper function with `is_inference`. We also have a separate section of the file for helpers (above line 1260) which would be a good place for this function.